### ISSUE TITLE
Feature request: Allow monochrome value assignment for SurfacePlot

### ISSUE THREAD

**Issue by michaelaye** [vispy/vispy#1004]
To assign a colormap to scene.visuals.SurfacePlot the `colors` parameter currently requires:
```python
colors : ndarray
    (width, height, 4) array of vertex colors.
```
The FR is that I would like to be able to just provide a monochrome array (width, height) instead of an array of (width, height, RGBA).

**Comment by campagnola** [comment]
This suggests two signatures for SurfacePlot:
```
SurfacePlot(z=array(w, h), color=array(w, h, 4))
SurfacePlot(z=array(w, h), color=array(w, h), cmap='...')
```
..but this is a very common theme across all visuals. I wonder if we can just generalize it like:
```
cmap = get_colormap('...')
color = cmap(array(w, h))
SurfacePlot(z=(w, h), color=color)
```
In this case, `color` would be something like a `MappedData` instance that represents raw data bound to a chain of functions, but not yet processed. It can be executed on the GPU, or on the CPU by calling `color.map()`.

The important advantage here is that we decouple all of the colormap / transform functionality from the APIs of the individual visuals.

Another advantage is that we also expose the chainability of transformations to the user:
```
from vispy.color importcubehelix, log, rescale
colors = cubehelix(log(rescale(array(w, h), -10, 10)))
SurfacePlot(z=(w, h), colors=colors)
```

**Comment by michaelaye** [comment]
Do I understand the last example correctly in that it doesn't actually do that calculation still until the colors object would call its method map()?

**Comment by campagnola** [comment]
That's right--calling `color.map()` would actually do the processing on the CPU, or passing it to the visual would cause it to be executed on the GPU.
